database/CoreDatabase.java

```diff
@@ -160,23 +160,27 @@ protected void initialise() {
 
     private void openSchema() {
         try {
-            List<ColumnFamilyDescriptor> schemaDescriptors = RocksPartitionManager.Schema.descriptors(rocksConfiguration.schema());
+            List<ColumnFamilyDescriptor> schemaDescriptors = CorePartitionManager.Schema.descriptors(rocksConfiguration.schema());
             List<ColumnFamilyHandle> schemaHandles = new ArrayList<>();
             rocksSchema = OptimisticTransactionDB.open(
                     rocksConfiguration.schema().dbOptions(),
                     directory().resolve(Encoding.ROCKS_SCHEMA).toString(),
                     schemaDescriptors,
                     schemaHandles
             );
-            rocksSchemaPartitionMgr = new RocksPartitionManager.Schema(schemaDescriptors, schemaHandles);
+            rocksSchemaPartitionMgr = partitionManagerSchema(schemaDescriptors, schemaHandles);
         } catch (RocksDBException e) {
             throw TypeDBException.of(e);
         }
     }
 
+    protected CorePartitionManager.Schema partitionManagerSchema(List<ColumnFamilyDescriptor> schemaDescriptors, List<ColumnFamilyHandle> schemaHandles) {
```

Comment:
Create a factory method that can be overridden by the subclass, providing their own schema partition manager implementation.